### PR TITLE
added supports_check_mode=True to info/facts modules

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Added `supports_check_mode=True` to `*_info` and `*_facts` modules.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
plugins/modules/cloud/alicloud/ali_instance_info.py
plugins/modules/cloud/memset/memset_memstore_info.py
plugins/modules/cloud/memset/memset_server_info.py
plugins/modules/cloud/rackspace/rax_facts.py
plugins/modules/cloud/smartos/smartos_image_info.py
plugins/modules/net_tools/snmp_facts.py
plugins/modules/remote_management/redfish/idrac_redfish_info.py
plugins/modules/remote_management/redfish/redfish_info.py
plugins/modules/system/xfconf_info.py
plugins/modules/web_infrastructure/sophos_utm/utm_aaa_group_info.py
plugins/modules/web_infrastructure/sophos_utm/utm_ca_host_key_cert_info.py
plugins/modules/web_infrastructure/sophos_utm/utm_network_interface_address_info.py

### CODE REVIEW

File: changelogs/fragments/3084-info-checkmode.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+minor_changes:
```
Comment (felixfontein) [inline changelogs/fragments/3084-info-checkmode.yaml:1]: These are clearly bugs.
```suggestion
bugfixes:
```

**Comment by felixfontein** [comment]
There are more `_info`/`_facts` modules which don't provide `supports_check_mode=True`:
```
ERROR: plugins/modules/cloud/misc/xenserver_facts.py:0:0: invalid-ansiblemodule-schema: AnsibleModule.supports_check_mode: required key not provided @ data['supports_check_mode']. Got None
ERROR: plugins/modules/remote_management/oneview/oneview_datacenter_info.py:0:0: invalid-ansiblemodule-schema: AnsibleModule.supports_check_mode: not a valid value for dictionary value @ data['supports_check_mode']. Got False
ERROR: plugins/modules/remote_management/oneview/oneview_enclosure_info.py:0:0: invalid-ansiblemodule-schema: AnsibleModule.supports_check_mode: not a valid value for dictionary value @ data['supports_check_mode']. Got False
ERROR: plugins/modules/remote_management/oneview/oneview_ethernet_network_info.py:0:0: invalid-ansiblemodule-schema: AnsibleModule.supports_check_mode: not a valid value for dictionary value @ data['supports_check_mode']. Got False
ERROR: plugins/modules/remote_management/oneview/oneview_fc_network_info.py:0:0: invalid-ansiblemodule-schema: AnsibleModule.supports_check_mode: not a valid value for dictionary value @ data['supports_check_mode']. Got False
ERROR: plugins/modules/remote_management/oneview/oneview_fcoe_network_info.py:0:0: invalid-ansiblemodule-schema: AnsibleModule.supports_check_mode: not a valid value for dictionary value @ data['supports_check_mode']. Got False
ERROR: plugins/modules/remote_management/oneview/oneview_logical_interconnect_group_info.py:0:0: invalid-ansiblemodule-schema: AnsibleModule.supports_check_mode: not a valid value for dictionary value @ data['supports_check_mode']. Got False
ERROR: plugins/modules/remote_management/oneview/oneview_network_set_info.py:0:0: invalid-ansiblemodule-schema: AnsibleModule.supports_check_mode: not a valid value for dictionary value @ data['supports_check_mode']. Got False
ERROR: plugins/modules/remote_management/oneview/oneview_san_manager_info.py:0:0: invalid-ansiblemodule-schema: AnsibleModule.supports_check_mode: not a valid value for dictionary value @ data['supports_check_mode']. Got False
ERROR: plugins/modules/web_infrastructure/sophos_utm/utm_proxy_frontend_info.py:0:0: invalid-ansiblemodule-schema: AnsibleModule.supports_check_mode: not a valid value for dictionary value @ data['supports_check_mode']. Got False
ERROR: plugins/modules/web_infrastructure/sophos_utm/utm_proxy_location_info.py:0:0: invalid-ansiblemodule-schema: AnsibleModule.supports_check_mode: not a valid value for dictionary value @ data['supports_check_mode']. Got False
```

**Comment by russoz** [comment]
Force pushed due to rebasing

**Comment by russoz** [comment]
> There are more `_info`/`_facts` modules which don't provide `supports_check_mode=True`:

@felixfontein done! Thanks for spotting those!

**Comment by felixfontein** [comment]
@russoz I didn't, but I wrote a sanity test which did ;-)

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks a lot for fixing this!

**Comment by felixfontein** [comment]
I created a manual backport to stable-2 in #3095. The backport contains a second commit which fixes more modules that were removed in c.g 3.0.0. (I think it's best to backport that commit to stable-1 and then add some more fixes if needed, so let's wait with that a bit...)

**Pull request merged**